### PR TITLE
Update auth block

### PR DESCRIPTION
**Pull request by akellbl4**
- API placed inside auth component
- Convert dropdown to hook (`setInterval` changed to `MutationObserver` for changing the height of the window)
- Removed `sleep` from auth block (other dropdowns should be changed in the same way)
- Auth has its own button (the current button should be replaced by it because the new one uses css-modules and rely on the global theme defined by global class name)

Tasks:
- [x]  New design looks like prototype #677 
- [x] Permanent token step (in case of miss click out of dropdown) #563
- [x] Merge separated code
- [x] Write tests
- [x] Fix translations
- [x] Finalize changes

### CODE REVIEW

**Comment by paskal** [comment]
I would propose doing "Magic login links" in a separate PR, it doesn't have to be tied to any other changes you want to introduce to the auth module. Blending it together only will make the review harder.

File: frontend/package.json [hunk 1]
```diff
@@ -26,91 +26,93 @@
...
+    "react-intl": "^5.12.5",
     "react-redux": "^7.2.2",
     "redux": "^4.0.5",
     "redux-thunk": "^2.3.0"
   },
   "devDependencies": {
-    "@babel/preset-env": "^7.12.11",
-    "@babel/preset-react": "^7.12.10",
-    "@formatjs/cli": "^3.0.1",
+    "@babel/preset-env": "^7.12.17",
```
Comment (paskal) [inline frontend/package.json:45]: Let's bump dependencies in a separate PR? It doesn't seem related to auth changes.

Comment (akellbl4) [reply frontend/package.json:45]: Sure, I've already made a separate brand. PR will be in a min.

**Comment by akellbl4** [comment]
@Mavrin hi there, I've finished writing tests. Could you please take a shallow look at it? Maybe you have some suggestions because I've done it the first time with `testing library`

File: frontend/app/custom-properties.css [hunk 2]
```diff
@@ -1,52 +1,50 @@
 :root {
-  --color0: #000;
-  --color1: #aaa;
-  --color3: #e2efef;
-  --color4: #edf6f7;
-  --color5: #eee;
-  --color6: #fff;
+  --color0: #0f172a;
```
Comment (akellbl4) [inline frontend/app/custom-properties.css:2]: Variables were just sorted by color.
CleanShot 2021-02-25 at 21 41 40@2x

File: frontend/app/components/auth/components/oauth-button.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import type { OAuthProvider } from 'common/types';
+import { siteId } from 'common/settings';
+import capitalizeFirstLetter from 'utils/capitalize-first-letter';
+import messages from 'components/auth/auth.messsages';
+
+import styles from './oauth-button.module.css';
+
+const OAUTH_ICONS = {
+  facebook: require('./assets/facebook.svg').default as string,
```
Comment (Mavrin) [inline frontend/app/components/auth/components/oauth-button.tsx:14]: Why do you use require instead of import?

File: frontend/app/components/auth/components/assets/microsoft.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg width="16" height="16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill="#EC4C23" d="M0 0h7.467v7.467H0z"/><path fill="#7DB300" d="M8.533 0H16v7.467H8.533z"/><path fill="#F7B302" d="M8.533 8.533H16V16H8.533z"/><path fill="#019FE8" d="M0 8.533h7.467V16H0z"/></svg>
```
Comment (Mavrin) [inline frontend/app/components/auth/components/assets/microsoft.svg:1]: end of line

**Review by Mavrin — CHANGES_REQUESTED** [review]

File: frontend/app/components/auth/components/oauth-button.tsx (hunk 3 above)
Comment (akellbl4) [reply frontend/app/components/auth/components/oauth-button.tsx:14]: In my opinion, it looks better in comparison with imports. I don't like the fence of the imports.
The problem with it is we should get `.default`. It could be solved with disabling esmodule for `.svg`. I haven't done it from the first attempt and have left as it is for a better time.

Both approaches are acceptable. It's all only about style in this case. For JS chunks it works in a slightly different way.

```js
const OAUTH_ICONS = {
  facebook: require('./assets/facebook.svg').default as string,
}
```

```js
import facebookIcon from './assets/facebook.svg'

const OAUTH_ICONS = {
  facebook: facebookIcon
}
```

File: frontend/app/components/auth/components/oauth-button.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  variant: 'full' | 'name' | 'icon';
+  onClick: JSX.GenericEventHandler<HTMLAnchorElement>;
+  provider: OAuthProvider;
+};
+
+const OAuthButton: FunctionComponent<OAuthIconProps> = ({ provider, onClick, variant }) => {
+  const intl = useIntl();
+
+  const [icon, setIcon] = useState('');
+  const providerName = capitalizeFirstLetter(provider);
```
[the remark this answers is not in the document]
Comment (akellbl4) [reply frontend/app/components/auth/components/oauth-button.tsx:35]: Yep, it looks reasonable. I tried to reduce hardcode but it doesn't work by capitalizing for this case.

**Review by Mavrin — APPROVED** [review]
